dataclasses_json/core.py

```diff
@@ -244,7 +244,16 @@ def _support_extended_types(field_type, field_value):
         res = (field_value
                if isinstance(field_value, UUID)
                else UUID(field_value))
-    elif _issubclass_safe(field_type, (int, float, str, bool)):
+    elif _issubclass_safe(field_type, bool):
+        # issubclass(bool, int) -> True
+        # thus values >1 will always convert to True in the next clause, unless intercepted
+        if field_value in ('True', 'true', True, 1):
+            res = True
+        elif field_value in ('False', 'false', False, 0):
+            res = False
```

Comment:
In JavaScript (where JSON is born out of), empty list, empty dict and empty string are all considered false, and if they just ahve one or more entry, they are true.

Reply:
Empty lists/dicts/strings are normally not decoded as booleans, even though in JS that is a common place, where JSON is used, this behaviour is avoided.

Reply:
As I do not like type juggling myself, I will rest this case

Reply:
Also, I was mistaken,  empty list and empty dict are coerced to true in JS